### ISSUE TITLE
headingLevelOffset is not considered if set in the .pillar file

### ISSUE THREAD

**Issue by VincentBlondeau** [pillar-markup/pillar#113]
Put "headingLevelOffset":2 metadescription in the .pillar do not work but it works in the pillar.conf.

It should be working in both

**Comment by YannDub** [comment]
I found the problem, when the headingLevelOffset transformation phase is launched, we have not yet the configuration of the file but pillar.conf in memory.

We need to parse the document to know its configuration, for the moment, we have no way to pass a configuration between two phases.

**Comment by DamienCassou** [comment]
@YannDub I don't understand your explanation. What do you mean by

> we have not yet the configuration of the file but pillar.conf in memory

? Moreover, from what I remember, all phases had access to the configuration.

**Comment by YannDub** [comment]
> @YannDub I don't understand your explanation. What do you mean by
> 

> > we have not yet the configuration of the file but pillar.conf in memory

The document configuration is computed when we export the document. When we are on the heading level offset transformation, we have the configuration of the pillar.conf and we use this configuration for all phases (except export phases).

> Moreover, from what I remember, all phases had access to the configuration.

Effectively we have access to the configuration, but I think the configuration of a document should be parse at the parsing phase (maybe at same time that the document). For the moment, a phase does not know the next phase and we can't get the configuration of a document to the next phases.

**Comment by YannDub** [comment]
Done on stable version, the file inclusion re execute all previous phases.
